XMonad/Prelude.hs

```diff
@@ -0,0 +1,54 @@
+{-# LANGUAGE BangPatterns #-}
+--------------------------------------------------------------------
+-- |
+-- Module      :  XMonad.Prelude
+-- Copyright   :  slotThe <soliditsallgood@mailbox.org>
+-- License     :  BSD3-style (see LICENSE)
+--
+-- Maintainer  :  slotThe <soliditsallgood@mailbox.org>
+--
+-- Utility functions and re-exports for a more ergonomic developing
+-- experience.  Users themselves will not find much use here.
+--
+--------------------------------------------------------------------
+module XMonad.Prelude (
+    module Exports,
+    fi,
+    chunksOf,
+    (.:),
+    (!?),
+) where
+
+import Control.Applicative as Exports
+import Control.Monad       as Exports
+import Data.Bool           as Exports
+import Data.Char           as Exports
+import Data.Foldable       as Exports
+import Data.Function       as Exports
+import Data.Functor        as Exports
+import Data.List           as Exports
+import Data.Maybe          as Exports
+import Data.Monoid         as Exports
+
+-- | Short for 'fromIntegral'.
+fi :: (Integral a, Num b) => a -> b
+fi = fromIntegral
+
+-- | Given a maximum length, splits a list into sublists
+--
+-- >>> chunksOf 5 (take 30 $ repeat 'a')
```

Comment:
I'm still trying to figure out if we want to depend on `split` for this instead of reinventing one function.

Reply:
I was thinking about this but ultimately decided against pulling in a dependency (even though `split` is really small) for a single function.  If we find more functions that are just taken from that package then I think the incentive would be much higher for us to just add it as a dependency

Reply:
Found one more: `X.P.Shell.split` is something like `Data.List.Split.splitOn . (:[])` with a special case for `[]`, returning `[]` instead of `[""]`.